### PR TITLE
Handling private issues

### PR DESCRIPTION
**Pull request by ralphbean**
https://issues.redhat.com/browse/RHTAP-1959

### CODE REVIEW

File: docs/modules/ROOT/pages/getting-started/getting_started_in_cli.adoc [hunk 1]
```diff
@@ -14,15 +14,19 @@ To log in to {ProductName} by using the CLI, complete the following steps:
...
 
 To verify that `oc login` worked correctly, complete the following steps:
 
 . Run `cd ~/.kube`.
-. Run `ls` and verify that there is a file named `config`. 
+. Run `ls` and verify that there is a file named `config`.
+
+== Switching workspaces
+
+In order to use the CLI tool with a workspace other than your default personal user workspace, append `/workspaces/<workspace-name>` to the AppStudio API url used in your `oc login` command.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/getting-started/getting_started_in_cli.adoc:29]:
```suggestion
To use the CLI tool with a workspace other than your default personal workspace, append `/workspaces/<workspace-name>` to the AppStudio API url used in your `oc login` command in step 6.
```

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 2]
```diff
@@ -1,85 +1,87 @@
...
+. In your team's normal workspace (not the private workspace), use the CLI to identify a `Snapshot` that you want to use as a basis for your new release. A common choice is "the most recently released Snapshot", which you can find by picking through your history of Releases sorted by creation date to find the relevant Snapshot identifier.
+. Use `oc` to extract that Snapshot to a local yaml file with `oc get snapshots <that snapshot id> -o yaml > snapshot.yaml`
+. Edit that file to construct a new `Snapshot` that replaces the impacted components pullspec with the pullspec of the image bearing the fix, pinned to digest. You can discard all annotations and labels and should give the Snapshot a new `name`.
+. Create that new Snapshot in the sytem with `oc apply -f snapshot.yaml`.
+. Watch your integration test pipelines confirm the `Snapshot` as valid and releasable.
+. Create a `Release` resource referencing that `Snapshot`. Follow a similar procedure as above. Export the old `Release` as a basis. Modify it to refer to the new `Snapshot`, etc.
+. In your team's normal workspace (not the private workspace), use the UI to watch the progress of the `Release`.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:83]:
```suggestion
. In your team's normal workspace (not the private workspace), use the UI to watch the progress of the `Release`. When completed, your fix is live and available to your customers.
```

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 3]
```diff
@@ -1,85 +1,87 @@
...
+. When it completes, your fix is now live and available to your customers.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:84]:
```suggestion

```

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 4]
```diff
@@ -1,85 +1,87 @@
...
+. Cherry pick the commit(s) from your private mirror back to the main git repository used by your team for normal development.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:85]:
```suggestion
. Cherry-pick the commit(s) from your private mirror back to the main Git repository used by your team for normal development.
```

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 5]
```diff
@@ -1,85 +1,87 @@
...
+
+WARN: Be sure to apply the patch(es) from your private mirror to your the git repository that your team normally uses for managing development. If you do not do this, then subsequent releases from your team will cause security regressions.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:87]:
```suggestion
IMPORTANT: Ensure to apply the patch(es) from your private mirror to the Git repository that your team normally uses for managing development. Failure to do so results in security regressions in subsequent releases from your team.
```

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 6]
```diff
@@ -1,85 +1,87 @@
...
+.**Procedure**
 
+. In the private workspace, use the UI to identify the pullspec of the image bearing the fix, pinned to digest.
+. In your team's normal workspace (not the private workspace), use the CLI to identify a `Snapshot` that you want to use as a basis for your new release. A common choice is "the most recently released Snapshot", which you can find by picking through your history of Releases sorted by creation date to find the relevant Snapshot identifier.
+. Use `oc` to extract that Snapshot to a local yaml file with `oc get snapshots <that snapshot id> -o yaml > snapshot.yaml`
+. Edit that file to construct a new `Snapshot` that replaces the impacted components pullspec with the pullspec of the image bearing the fix, pinned to digest. You can discard all annotations and labels and should give the Snapshot a new `name`.
+. Create that new Snapshot in the sytem with `oc apply -f snapshot.yaml`.
+. Watch your integration test pipelines confirm the `Snapshot` as valid and releasable.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:81]:
```suggestion
. Watch your integration test pipelines to confirm that the `Snapshot` is valid and releasable.
```

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 7]
```diff
@@ -1,85 +1,87 @@
...
-----
-oc apply -f <path-to-yaml>
-----
-
-. Within the RHTAP UI, import the impacted component into the same application that the user has just imported. This ensures that the application is fully functional in the secure workspace.
+.**Procedure**
 
+. In the private workspace, use the UI to identify the pullspec of the image bearing the fix, pinned to digest.
+. In your team's normal workspace (not the private workspace), use the CLI to identify a `Snapshot` that you want to use as a basis for your new release. A common choice is "the most recently released Snapshot", which you can find by picking through your history of Releases sorted by creation date to find the relevant Snapshot identifier.
+. Use `oc` to extract that Snapshot to a local yaml file with `oc get snapshots <that snapshot id> -o yaml > snapshot.yaml`
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:78]:
```suggestion
. Use `oc` to extract the Snapshot that you want to use as a basis for your new release to a local yaml file with `oc get snapshots <that snapshot id> -o yaml > snapshot.yaml`
```

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 8]
```diff
@@ -1,85 +1,87 @@
...
+. Edit that file to construct a new `Snapshot` that replaces the impacted components pullspec with the pullspec of the image bearing the fix, pinned to digest. You can discard all annotations and labels and should give the Snapshot a new `name`.
+. Create that new Snapshot in the sytem with `oc apply -f snapshot.yaml`.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:80]:
```suggestion
. Create that new Snapshot in the system with `oc apply -f snapshot.yaml`.
```

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 9]
```diff
@@ -1,85 +1,87 @@
...
-
-+
-[source,bash]
-----
-./rhtap export application <name_of_your_application>  --from <original-tenant-workspace> --to <secure-workspace> --as-prebuilt-images --skip <impacted component>
-----
-
-+
-NOTE: Ensure that you have excluded the impacted component from which you have created a private mirror in step 1.
+. Verify that pull requests to your private mirror create pipelineruns in the private workspace, and that integration tests run correctly.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:61]:
```suggestion
. Verify that pull requests to your private mirror create PipelineRuns in the private workspace and that integration tests run correctly.
```

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 10]
```diff
@@ -1,85 +1,87 @@
...
 
-.**Prerequisite**
++
+[source,bash]
+----
+oc apply -f ./data/<path-to-yaml>
+----
 
-* You are the maintainer of the application in the original workspace where the embargoed issue was identified.
+. Within the RHTAP UI, in the private workspace, import the private mirror as a new component in the application that was just created by the `oc` step above.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:49]:
```suggestion
. Within the RHTAP UI, in the private workspace, import the private mirror as a new component in the application that you just created in step 7.
```

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 11]
```diff
@@ -1,85 +1,87 @@
...
 
-* You have downloaded RHTAP link:https://github.com/redhat-appstudio/rhtap-cli/releases[pre-built CLI].
+At this point, you should have a clone of your application in the new private workspace with all of its components mirrored from the origin workspace, except the impacted component which is bound to the new private git mirror repository.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:51]:
```suggestion
At this point, you should have a clone of your application in the new private workspace, with all of its components mirrored from the origin workspace, except the impacted component, which is now linked to the new private Git mirror repository.
```

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 12]
```diff
@@ -1,85 +1,86 @@
...
+. Use `oc` to extract the Snapshot that you want to use as a basis for your new release to a local yaml file with `oc get snapshots <that snapshot id> -o yaml > snapshot.yaml`
+. Edit the extracted file to construct a new `Snapshot` that replaces the impacted components pullspec with the pullspec of the image bearing the fix, pinned to digest. You can discard all annotations and labels and should give the Snapshot a new `name`.
+. Create that new Snapshot in the system with `oc apply -f snapshot.yaml`.
+. Review your integration test pipelines to confirm that the `Snapshot` is valid and can be released.
+. Create a `Release` resource referencing the specified `Snapshot`. Follow a similar procedure as outlined above. Export the existing `Release` as a starting point. Modify it to reference the new `Snapshot` and make any necessary adjustments.
+. In your team's normal workspace (not the private workspace), use the UI to watch the progress of the `Release`. When completed, your fix is live and available to your customers.
+. Cherry-pick the commit(s) from your private mirror back to the main Git repository used by your team for normal development.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:84]: This won't necessarily be possible/easy as there is no link between the repos. The commit will have to be ported over.

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 13]
```diff
@@ -1,85 +1,86 @@
...
+
+IMPORTANT: Ensure to apply the patch(es) from your private mirror to the Git repository that your team normally uses for managing development. Failure to do so results in security regressions in subsequent releases from your team.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:86]: We still do not address the need to keep the rest of the components up to date in the private workspace to ensure proper functionality in the broader sense. Should we call this out?

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 14]
```diff
@@ -1,85 +1,86 @@
...
+[source,bash]
+----
+oc apply -f ./data/<path-to-yaml>
+----
 
-* You are the maintainer of the application in the original workspace where the embargoed issue was identified.
+. Within the RHTAP UI, in the private workspace, import the private mirror as a new component in the application that you just created in step 7.
 
-* You have downloaded RHTAP link:https://github.com/redhat-appstudio/rhtap-cli/releases[pre-built CLI].
+At this point, you should have a clone of your application in the new private workspace, with all of its components mirrored from the origin workspace, except the impacted component, which is now linked to the new private Git mirror repository.
```
[the remark this answers is not in the document]
Comment (ralphbean) [reply docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:51]: Oh, yes.

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 15]
```diff
@@ -1,85 +1,86 @@
...
-. To clone the application from one workspace to another, run the following command:
+. Create a link:https://docs.github.com/en/repositories/creating-and-managing-repositories/duplicating-a-repository[Private Mirror] of the corresponding repository for any components that will receive fixes related to embargoed information. This step is crucial to safeguard sensitive data and prevent unauthorized access.
+
++
+NOTE: Creating a private fork of the public link:https://docs.github.com/en/repositories/creating-and-managing-repositories/duplicating-a-repository[GitHub repository] is not possible.
+
+. link:https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/managing-workspaces/proc_creating_a_team_workspace/[Submit a request for a new workspace] dedicated to work on the embargoed issue. This workspace will serve as the private environment for cloning the affected application. Don't expose information about the vulnerability in your choice of a name for the workspace.
```
[the remark this answers is not in the document]
Comment (ralphbean) [reply docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc]: Sorry, that was 3ab3677.

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc (hunk 14 above)
[the remark this answers is not in the document]
Comment (ralphbean) [reply docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc]: See 16d9c8f

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc (hunk 12 above)
Comment (arewm) [reply docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:84]: We can leave it out of scope if we use different language, i.e.
```suggestion
. Replicate the commit(s) from your private mirror back to the main Git repository used by your team for normal development.
```

or other words like "migrate"

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc (hunk 13 above)
Comment (ralphbean) [reply docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:86]: I added a note in 361c86b. WDYT?

**Review by gtrivedi88 — APPROVED** [review]

File: docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc [hunk 16]
```diff
@@ -1,85 +1,88 @@
...
+* **Prepare the fix:** Preparing the fix with source code in your private git mirror and builds in your private workspace.
+* **Release the fix:** Releasing the build from your private workspace and "upstreaming" your patch from your private git mirror back to your team's normal development repo.
 
-* **Re-create application with limited visibility:** Re-creating applications while minimizing the awareness of their existence, ensuring a secure environment for working on security fixes.
-
-
-== Cloning applications to a dedicated workspace
-You can clone applications to a dedicated workspace to address security vulnerabilities, test new features, and enhance the application without disrupting the production environment. 
+== Cloning applications to a private workspace
+Cloning an application involves replicating the application's code and configuration into a dedicated private workspace, ensuring the sensitive information remains isolated and protected.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc:12]:
```suggestion
Cloning an application involves replicating the application's code and configuration into a dedicated private workspace, ensuring that sensitive information remains isolated and protected.
```

**Pull request merged**